sdk/eventgrid/azure-eventgrid/migration_guide.md

```diff
@@ -0,0 +1,84 @@
+
+### Cross Service SDK improvements
+
+The modern Event Grid client library also provides the ability to share in some of the cross-service improvements made to the Azure development experience, such as 
+- a unified logging and diagnostics pipeline offering a common view of the activities across each of the client libraries
+
+## Important changes
+
+### Support for Cloud Events
+
+The v2.x major version comes with support for [CloudEvents](https://github.com/cloudevents/spec). Now the cloud native Cloud Events can be directly published using the `CloudEvent` constructor or a as a dictionary as follows:
+
+```Python
+from azure.eventgrid import CloudEvent
+
+cloud_event = CloudEvent(
+    type="Contoso.Items.ItemReceived",
+    source="/contoso/items",
+    data={
+        "itemSku": "Contoso Item SKU #1"
+    },
+    subject="Door1"
+)
+
+# as a dictionary
+
+cloud_event = {
+    "type":"a0517898-9fa4-4e70-b4a3-afda1dd68672",
+    "source":"/subscriptions/{subscription-id}/resourceGroups/{resource-group}/providers/Microsoft.Storage/storageAccounts/{storage-account}",
+    "data": {"hello": "world"},
+    "subject": "Door1"
+    }
+```
+
+### Client constructors
+
+* The `EventGridClient` in the v1.3 has been replaced with `EventGridPublisherClient`.
+
+| In v1.3 | Equivalent in v2.0 | Sample |
+|---|---|---|
+|`EventGridClient(credentials)`|`EventGridPublisherClient(topic_hostname, credential)`|[Sample for client construction](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/eventgrid/azure-eventgrid/samples/champion_scenarios/cs5_publish_events_using_cloud_events_1.0_schema.py)|
+
+* Additionally, we now have an `EventGridConsumer` that should be used to deserialize the events.
+
+```Python
+from azure.eventgrid import EventGridConsumer
+
+eg_consumer = EventGridConsumer()
+```
+
+### Publishing Events
+
+The `publish_events` API is replaced with `send` in v2.0. Additionally, `send` API accepts `CloudEvent`, `CustomEvent` along with `EventGridEvent`
+
+| In v1.3 | Equivalent in v2.0 | Sample |
+|---|---|---|
+|`EventGridClient(credentials).publish_events(topic_hostname, events)`|`EventGridPublisherClient(topic_hostname, credential).send(events)`|[Sample for client construction](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/eventgrid/azure-eventgrid/samples/champion_scenarios/cs5_publish_events_using_cloud_events_1.0_schema.py)|
+
+### Consuming Events
+
+In v2.0, `EventGridConsumer` can be used to decode both Cloud Events and EventGrid Events.
+
+```Python
+EventGridConsumer().decode_cloud_event(cloud_event)
+
+EventGridConsumer().decode_eventgrid_event(eventgrid_event)
```

Comment:
I'm wondering how the cloud_event and eventgrid_event are received, do you think it would be helpful to demonstrate how to receive the events as well?

Reply:
i added a link for samples instead as this is just a migration guide